File: order-service/src/main/java/com/caglar/order/service/AdminService.java
package com.caglar.order.service;

import com.caglar.order.dto.response.AdminStatsResponseDto;
import com.caglar.order.dto.response.OrderResponseDto;
import com.caglar.order.enums.OrderStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface AdminService {
    AdminStatsResponseDto stats();
    Page<OrderResponseDto> orders(OrderStatus status, Pageable pageable);
    OrderResponseDto updateStatus(Long orderId, OrderStatus newStatus);
}

File: n11-client/src/services/admin.service.ts
import { http } from "./http";
import type { AdminStats, Order, OrderStatus, Page } from "@/types/api";

export const adminService = {
  stats: () => http.get<AdminStats>("/admin/stats"),
  orders: (status?: OrderStatus, page = 0, size = 20) =>
    http.get<Page<Order>>("/admin/orders", { query: { status, page, size } }),
  updateOrderStatus: (id: number, status: OrderStatus) =>
    http.post<Order>(`/admin/orders/${id}/status`, undefined, { query: { status } }),
};
